src/mlpack/methods/ann/dists/normal_distribution_impl.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file normal_distribution.cpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "normal_distribution.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution() :
+    applySoftplus(true)
+{
+  // Nothing to do here.
+}
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution(
+    const DataType&& mean,
+    const DataType&& stdDev) :
+    mean(mean),
+    stdDev(stdDev)
+{
+  if (mean.size() != stdDev.size())
+  {
+    Log::Fatal << "NormalDistribution<>::NormalDistribution(): The sizes of "
+        << "the mean and the standard deviation should be equal." << std::endl;
+  }
+}
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution(
+    const DataType&& param,
+    const bool applySoftplus) :
+    mean(param.submat(param.n_rows / 2, 0, param.n_rows - 1,
+        param.n_cols - 1)),
+    preStdDev(param.submat(0, 0, param.n_rows / 2 - 1, param.n_cols - 1)),
+    applySoftplus(applySoftplus)
+{
+  if (param.n_rows % 2 != 0)
+  {
+    Log::Fatal << "NormalDistribution<>::NormalDistribution(): The number of "
+        << "rows of param matrix should be even." << std::endl;
+  }
+
+  if (applySoftplus)
+    SoftplusFunction::Fn(preStdDev, stdDev);
+  else
+    stdDev = preStdDev;
```

Comment:
Is this the only place `preStdDev` is used?  If so we can probably remove it as a class member and only use it locally here.

Reply:
Ah, sorry, in this case you can ignore my comment. :)